salt/migrations/2-3/haproxy/init.sls

```diff
@@ -0,0 +1,65 @@
+include:
+  - ca-cert
+  - cert
+
+# NOTE: Remove me for 4.0
+
+/etc/caasp/haproxy/haproxy.cfg:
+  file.managed:
+    - source: salt://haproxy/haproxy.cfg.jinja
+    - template: jinja
+    - user: root
+    - group: root
+    - mode: 644
+    - makedirs: True
+    - dir_mode: 755
+
+{% from '_macros/certs.jinja' import certs, alt_master_names with context %}
+{{ certs("kube-apiserver-proxy",
+         pillar['ssl']['kube_apiserver_proxy_crt'],
+         pillar['ssl']['kube_apiserver_proxy_key'],
+         cn = grains['nodename'] + '-proxy',
+         o = pillar['certificate_information']['subject_properties']['O'],
+         extra_alt_names = alt_master_names()) }}
+
+haproxy:
+  file.managed:
+    - name: /etc/kubernetes/manifests/haproxy.yaml
+    - source: salt://haproxy/haproxy.yaml.jinja
```

Comment:
Here we are using the `haproxy.yaml` file of the "main/upstream" HAProxy status on v2 nodes. Wouldn't it be safer to copy the current version into this  `salt/migrations/2-3/haproxy/` directory to avoid breakages in the future?